lms/services/application_instance.py

```diff
@@ -157,10 +165,42 @@ def create_application_instance(  # pylint:disable=too-many-arguments
 
         return application_instance
 
+    def update_from_lti_params(self, application_instance, params: LTIParams):
+        """
+        Update all the LMS-related attributes present in `params`.
+
+        If the current instance already has a `tool_consumer_instance_guid`
+        report it on logging and don't update any of the columns.
+        """
+
+        tool_consumer_instance_guid = params.get("tool_consumer_instance_guid")
+        if not tool_consumer_instance_guid:
+            # guid identifies the rest of the LMS data, if not there skip any
+            # updates
+            return
+
+        application_instance.check_guid_aligns(tool_consumer_instance_guid)
+
+        # Make sure this application instance is associated with an org.
+        self._organization_service.auto_assign_organization(application_instance)
```

Comment:
This addition here is why this had to move, as we need more info that we can get from looking at the instance itself

Reply:
Yeah I think having it here makes sense and we don't have to think that much about where to put things in the future for any model.